assets/blocks/editor-components/limited-text-control/index.test.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * External dependencies
+ */
+import { render, fireEvent } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import LimitedTextControl from './index';
+import userEvent from '@testing-library/user-event';
+
+describe( '<LimitedTextControl />', () => {
+	it( 'Should render the input with the correct value', () => {
+		const { queryByRole } = render(
+			<LimitedTextControl value="SOME_VALUE" maxLength={ 20 } />
+		);
+
+		expect( queryByRole( 'textbox' ) ).toHaveValue( 'SOME_VALUE' );
+	} );
+
+	it( 'Should render label', () => {
+		const { queryByLabelText } = render(
+			<LimitedTextControl
+				label="SOME_LABEL"
+				value="SOME_VALUE"
+				maxLength={ 20 }
+			/>
+		);
+
+		expect( queryByLabelText( 'SOME_LABEL' ) ).toHaveValue( 'SOME_VALUE' );
+	} );
+
+	it( 'Should call the change event', () => {
+		const onChangeMock = jest.fn();
+		const { queryByRole } = render(
+			<LimitedTextControl
+				value="SOME_VALUE"
+				onChange={ onChangeMock }
+				maxLength={ 20 }
+			/>
+		);
+
+		fireEvent.change( queryByRole( 'textbox' ), {
+			target: { value: 'SOME_OTHER_VALUE' },
+		} );
+
+		expect( onChangeMock ).toBeCalledWith( 'SOME_OTHER_VALUE' );
+	} );
+
+	it( 'Should display the correct character count', () => {
+		const { queryByText } = render(
+			<LimitedTextControl value="SOME_VALUE" maxLength={ 20 } />
+		);
+
+		expect( queryByText( 'Characters: 10/20' ) ).toBeTruthy();
+	} );
+
+	it( 'Should respect the maxLength property', async () => {
+		const onChangeMock = jest.fn();
+		const { queryByRole } = render(
+			<LimitedTextControl
+				value="ABC"
+				maxLength={ 3 }
+				onChange={ onChangeMock }
+			/>
+		);
+
+		await userEvent.type( queryByRole( 'textbox' ), 'DEF' );
+
+		expect( onChangeMock ).toHaveBeenCalledTimes( 0 );
```

Comment:
Thinking a little about this, I'd use a maxLength as `4`, and `toHaveBeenCalledTimes( 1 );`. So I think it's a little more tested. Otherwise, if `onChangeMock` is never called because it's not working, the test would pass. Does it make sense?

If you agree, I think the same applies to some of the other tests.

Reply:
I see your point, but I think we cover that case (when onChangeMock is never called) in the test "Should call the change event".